### PR TITLE
fix: remove defaults for CODER_EMAIL_ options

### LINKED ISSUE #15480: `CODER_NOTIFICATIONS_EMAIL_SMARTHOST`'s value is never used
`CODER_NOTIFICATIONS_EMAIL_SMARTHOST` is now deprecated, and superseded by `CODER_EMAIL_SMARTHOST` in https://github.com/coder/coder/pull/15270.

However, if `CODER_EMAIL_SMARTHOST` is unset then the value from `CODER_NOTIFICATIONS_EMAIL_SMARTHOST` should be used. Currently it is not, and I suspect it's because `CODER_EMAIL_SMARTHOST` has a default value:

```go
	emailSmarthost := serpent.Option{
		Name:        "Email: Smarthost",
		Description: "The intermediary SMTP host through which emails are sent.",
		Flag:        "email-smarthost",
		Env:         "CODER_EMAIL_SMARTHOST",
		Default:     "localhost:587", // To pass validation.
		Value:       &c.Notifications.SMTP.Smarthost,
		Group:       &deploymentGroupEmail,
		YAML:        "smarthost",
	}
```

### PR DESCRIPTION
**Pull request by DanielleMaywood**
Fixes https://github.com/coder/coder/issues/15480

The defaults in `$CODER_EMAIL_SMARTHOST`, `$CODER_EMAIL_HELLO`, and `$CODER_EMAIL_FORCE_TLS` override values set with the (now deprecated) `$CODER_NOTIFICATIONS_EMAIL_SMARTHOST`, `$CODER_NOTIFICATIONS_EMAIL_HELLO`, and `$CODER_NOTIFICATIONS_EMAIL_FORCE_TLS` environment variables.

This fixes that by removing the defaults. The defaults are helpful for local dev setups but for any serious deployment these will likely be changed anyways so this _shouldn't_ cause any issues for customers existing deployments.

### CODE REVIEW

**Comment by DanielleMaywood** [comment]
If https://github.com/coder/serpent/pull/23 gets merged (and released), this would fix the issue encountered here.

File: coderd/notifications/dispatch/smtp.go [hunk 1]
```diff
@@ -521,15 +520,13 @@ func (s *SMTPHandler) validateToAddrs(to string) ([]string, error) {
 // Does not allow overriding.
 // nolint:revive // documented.
 func (s *SMTPHandler) smarthost() (string, string, error) {
-	host := s.cfg.Smarthost.Host
-	port := s.cfg.Smarthost.Port
-
-	// We don't validate the contents themselves; this will be done by the underlying SMTP library.
-	if host == "" {
-		return "", "", ValidationNoSmarthostHostErr
+	if s.cfg.Smarthost.String() == "" {
```
Comment (dannykopping) [inline coderd/notifications/dispatch/smtp.go:523]: We may want to trim spaces as well to be sure.

**Review by dannykopping — APPROVED** [review]
Thanks for this! A couple minor points but otherwise LGTM.

File: codersdk/deployment.go [hunk 2]
```diff
@@ -1038,7 +1037,6 @@ when required by your organization's security policy.`,
 		Description: "The hostname identifying the SMTP server.",
 		Flag:        "email-hello",
 		Env:         "CODER_EMAIL_HELLO",
-		Default:     "localhost",
```
[the remark this answers is not in the document]
Comment (DanielleMaywood) [reply codersdk/deployment.go:1041]: I'm happy to do this although an issue with keeping this one is if someone has set `CODER_NOTIFICATIONS_EMAIL_HELLO` then we will be changing it back to "localhost" for them.

[the remark this answers is not in the document]
Comment (dannykopping) [reply codersdk/deployment.go:1041]: If we wait until https://github.com/coder/serpent/pull/23 is merged then that won't be the case, right?

[the remark this answers is not in the document]
Comment (DanielleMaywood) [reply codersdk/deployment.go:1041]: correct

[the remark this answers is not in the document]
Comment (mafredri) [reply codersdk/deployment.go:1041]: I managed to solve the only open question I had for coder/serpent#23 and I'm pretty happy with it. We've increased test coverage and all coder/coder tests are passing.

Basically just waiting for an approve, so do you want to go ahead with this PR @DanielleMaywood or do we go with coder/serpent#23? I didn't review yet, but if you want this merged I can do a quick review.

[the remark this answers is not in the document]
Comment (DanielleMaywood) [reply codersdk/deployment.go:1041]: I think we go with https://github.com/coder/serpent/pull/23 first, as that is a prerequisite to get this to behave how we want.

[the remark this answers is not in the document]
Comment (dannykopping) [reply codersdk/deployment.go:1041]: Looks like you have two reviewers for that PR already, but do shout if need another 👍

**Comment by DanielleMaywood** [comment]
Closing as scope changed. https://github.com/coder/coder/pull/15476 has fixed the original issue here (by updating to a new version of coder/serpent).

**Pull request closed without merging**